main.nf

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env nextflow
+
+
+// check parameters
+if (!params.inputDir) {
+    exit 1, "inputDir parameter is missing."
+}
+if (!params.outputDir) {
+    exit 1, "outputDir parameter is missing."
+}
+if ( !new File(params.inputDir).exists()) {
+    exit 1, "The input folder does not exists."+params.inputDir+"\n"
+}
+if (!params.outPrefix) {
+    exit 1, "Output prefix parameter is missing."
+}
+if (!params.debug) {
+    exit 1, "Debug prefix parameter is missing."
+}
+if (!params.localOutDir) {
+    params.localOutDir='bam'
+}
+
+pairFiles_ch = Channel.fromFilePairs( params.inputDir+"/*{1,2}.fastq.gz", size: 2, checkIfExists: true )
+
+process ERValign {
+    // tag ${sample}
+    
+    // executor configuration
+    time '8h'
+    // memory '35.GB'
+    scratch true
+    storeDir params.outputDir
+
```

Comment:
Since you pass the number of CPUs to STAR and to samtools, it would be a good idea to add the directive:
cpus $cpus
to this process.
See https://www.nextflow.io/docs/latest/process.html#cpus

Reply:
If I do this, do I need to pass cpus in the `input:` section? 
Can I still use `!{cpus}` in ERValign.sh ?

Reply:
You already have the cpus at line https://github.com/eipm/ERVmap/blob/b033035fed058cb0d0efb6e4d91fe3489df72228/main.nf#L42

As for sample, you can just use the variable in the directive.